packages/components/filters/src/filter-menu/trigger-button/trigger-button.tsx

```diff
@@ -1,8 +1,63 @@
-import { type ReactNode, forwardRef, type LegacyRef } from 'react';
+import {
+  type ReactNode,
+  forwardRef,
+  type LegacyRef,
+  useCallback,
+  useRef,
+  useState,
+} from 'react';
 import { CaretDownIcon, CloseBoldIcon } from '@commercetools-uikit/icons';
 import SecondaryIconButton from '@commercetools-uikit/secondary-icon-button';
 import { type TAppliedFilterValue } from '../filter-menu';
 import * as styles from './trigger-button.styles';
+import { Badge } from '../../badge';
+import { Chip } from '../chip';
+
+const useScrollObserver = (totalCount: number) => {
+  const [isOverflowing, setIsOverflowing] = useState(false);
+  const [visibleCount, setVisibleCount] = useState(0);
+  const containerRef = useRef<HTMLUListElement | null>(null);
+
+  const setContainerRef = useCallback(
+    (node: HTMLUListElement | null) => {
+      if (node) {
+        containerRef.current = node;
+
+        setTimeout(() => {
+          const { clientWidth, scrollWidth, children } = node;
+
+          const hasOverflow = scrollWidth > clientWidth;
+          setIsOverflowing(hasOverflow);
+
+          let widthOfAllChildren = 0;
+          let visibleChipCount = 0;
+
+          if (hasOverflow) {
+            Array.from(children).forEach((child, idx) => {
+              const { width } = (child as HTMLElement).getBoundingClientRect();
+              widthOfAllChildren += width + 4;
```

Comment:
Adding 4 to each child `width` since the width doesn't take the 4px gap into account

Reply:
Can you try calculating if the boundaries of the current child are within the boundaries of the container instead? Something along this:

```
const childRect = child.getBoundingClientRect();
const containerRect = containerRef.current.getBoundingClientRect();
if (
        childRect.left >= containerRect.left &&
        childRect.right <= containerRect.right
) {
        visible += 1; 
}
```

This way the *current* styling is not tied to the counter implementation.

Reply:
done in [this commit](https://github.com/commercetools/ui-kit/pull/2964/commits/11f41ac64e8c912ffe2ddfba53a0e4fe52686b46)